### DISCUSSION TITLE
OMI glTF Working Group Meeting 2022-05-26

### DISCUSSION THREAD

**Discussion by robertlong** [omigroup/gltf-extensions#86] [General]
# OMI glTF Working Group Meeting 5/26/2022

This meeting will be on 5/26/2022 at 10:30 PM (UTC) / 2:30 PM (PST) in the [OMI Discord](https://discord.gg/NJtT9grz5E) within the #omi-weekly-meeting channel. During the meeting, we will be using the #omi-gltf-extensions channel to manage a speaker queue, post links, and for any sidebar discussions.

To be notified of this meeting and others, subscribe to the [OMI Meetings and Events Calendar](https://calendar.google.com/calendar/u/1?cid=Y18wZHB1Z2Y5ZjgzZXE0cWVrbWI2b21xYmptZ0Bncm91cC5jYWxlbmRhci5nb29nbGUuY29t) or add yourself to the @omi-gltf-subgroup role in the #roles channel of the OMI Discord.

## Agenda:

- KHR_audio feedback
  - https://github.com/KhronosGroup/glTF/pull/2137#discussion_r871810364
  - Hubs: https://github.com/mozilla/hubs/pull/4741
- OMI_collider
- OMI_id / [OMI_ref](https://github.com/omigroup/gltf-extensions/pull/85)

To propose another item for the agenda, comment below (preferably before the meeting).

**Comment by antpb** [comment]
If we have time maybe we can discuss how I was using the media frame components from Hubs. I suspect there is some spec to be built from here: https://twitter.com/antpb/status/1529499461294096386?s=20&t=dEeTicdrshR9wKw-8tJyDg
<img width="580" alt="Screen Shot 2022-05-26 at 4 19 25 PM" src="https://user-images.githubusercontent.com/8985705/170581938-eeba75bf-bd45-4291-98bc-13402b34b591.png">

**Comment by fire** [comment]
We merged https://github.com/godotengine/godot/pull/57660 Improve GLTF Document extensions into Godot Engine.

We have also updated the godot VRM addon and a sample gltf sandbox.

Summary: Godot has a improved system for modifying the gltf importer/exporter via scripting plugins. Current state can now begin implementing the audio extensions.

## References

* https://github.com/V-Sekai/godot-vrm
* https://github.com/V-Sekai/godot-gltf-sandbox

**Comment by robertlong** [comment]
# Notes
- KHR_audio
  - Hubs seems to be doing a refactor to integrate an ECS system and has delayed merging our KHR_audio PR. However, they seem pretty supportive of it, which is great news!
- OMI_ref
  - We did some brainstorming on how references and overrides might work.
  - Perhaps you can only reference properties of equal types. In other words, the object with the `OMI_ref` must have the same type of the object you are referencing. Ex. a node can reference another node, but it can't reference a scene or material.
  - Overriding properties on the referenced object got a little more complicated:
    - Overriding properties on the directly referenced object is pretty simple. You can merge the two objects together. So a node with a transform will override the transform of the node it references.
    - Another way we could approach this is to allow exposing specific properties to be overridden. This would work for even deeply nested properties.
- @antpb showed off using Spoke's media frames as references to objects which can be replaced other assets via scripting.
  - Perhaps we could get the ability to define `node.extras` data in the Spoke properties panel. This would allow for a more direct way to reference objects via scripts and embed custom application data on a node.
- @fire showed off their latest work on Godot's glTF import/export plugin API. You can now import some of the Mozilla Hubs components into Godot and there's now some APIs to define your own importer/exporter extensions. We're not far from being able to create interoperable assets with custom extensions in Godot!
